onnxruntime/core/providers/cuda/tensor/trilu_impl.cu

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "trilu_impl.h"
+#include <stdio.h>
+namespace onnxruntime {
+namespace cuda {
+
+template <typename T>
+__global__ void TriluKernel(
+    bool upper,
```

Comment:
bool upper can be a template variable. 
It will save some computation in the kernel.